pyccel/parser/semantic.py

```diff
@@ -2166,14 +2242,15 @@ def _visit_FunctionalFor(self, expr, **settings):
         d_var['shape'] = PythonTuple(*shape)
         d_var['is_stack_array'] = False # PythonTuples can be stack arrays
 
-        lhs_name = _get_name(expr.lhs)
-
-        lhs_empty = Empty(shape, dtype = dtype)
-        lhs       = self._create_variable(lhs_name, dtype, lhs_empty, d_var)
+        # ...
+        # TODO [YG, 30.10.2020]:
+        #  - Check if we should use self._create_variable or self._assign_lhs_variable
+        #  - Check if we should allow the possibility that is_stack_array=True
+        #  - Verify that the LHS variable did not exist already
```

Comment:
Shouldn't we create issues rather than hiding TODOs in the code? If not then we probably need an issue about finding TODOs ^^'

Reply:
I am using these TODO items as a reference for discussion, which is exactly what it happening now :-). And they help me to keep track of the progress in this PR, given that it is still is a "draft".

I am not a big fan of creating lots of issues before a proper discussion. With a bit of luck and commitment we may be able to tackle all of these TODOs within this PR, without creating any new issue.

Reply:
This is `_visit_FunctionalDef` and it had problems well before this PR, so it's probably better to create an issue for it. 

Maybe @saidctb wants to look into this first?

Reply:
@saidctb The one TODO item that remains open is this one in `_visit_FunctionalDef`:
```python
# Check if we should allow the possibility that is_stack_array=True
```
what do you think about this?